src/lib.rs

```diff
@@ -408,61 +408,155 @@
+//! `long_help(<whole comment>)` so clap will display the summary with `-h`
+//! and the whole help message on `--help` (see below).
+//!
+//! So, the example above will be turned into this (details omitted):
+//! ```
+//! clap::App::new("<name>")
+//!     .about("Hi there, I'm Robo!")
+//!     .long_about("Hi there, I'm Robo!\n\n\
+//!                  I like beeping, stumbling, eating your electricity,\
+//!                  and making records of you singing in a shower.\
+//!                  Pay up or I'll upload it to youtube!")
+//! // args...
+//! # ;
+//! ```
+//!
+//! ### `-h` vs `--help` or `help()` vs `long_help()`
+//!
+//! `-h` flag is not the same as `--help`. `-h` corresponds to
+//! [`Arg::help`]/[`App::about`], it's for short "summary" messages.
+//! `--help` is for more detailed, descriptive messages,
+//! [`Arg::long_help`]/[`App::long_about`].
+//!
+//! It is entirely up to `clap` what happens if you used only one of
+//! `help`/`long_help`, see `clap`'s documentation for these methods.
+//!
+//! TL;DR: Current behavior (`clap v2.33`) is: if only one of these methods was
+//! called the same message will be displayed for *both* `-h` and `--help`. The
+//! same logic applies to `about/long_about`.
+//!
+//! ### Doc comment preprocessing and `#[structopt(no_mangle_doc)]`
+//!
+//! `structopt` applies some kind of preprocessing to doc comments to
+//! keep common usage convenient.
+//!
+//! What it does, exactly:
+//!
+//! * First of all, it *removes one leading space* from every line, if present.
+//!   We apply this to *all* doc comments, even if `#[structopt(no_mangle_doc)]`
+//!   is present (putting a space after `///` is common practice in Rust).
+//!
+//! * For non-"long" doc comments (second line is not whitespace only
+//!   or the whole comment is one line):
+//!     * Trim leading and trailing whitespace from each line
+//!     * Join all the lines, separating them by a space ` `
+//!     * Remove trailing dot `.` but only if it's only one dot
+//!       (i.e `structopt` erases periods but doesn't touch ellipses).
+//!
+//! * For "long" doc comments (second line is empty):
+//!     * Title is processed as non-"long" comment
+//!     * Detailed message (2+ lines) are processed just as non-"long" comments
+//!       except we don't remove trailing `.`
+//!     * Processed title and detailed message are concatenated as following:
+//!       `<title> + "\n\n" + <detailed>`
+//!
+//! Sometimes you don't want this preprocessing to apply, for example comment contains
+//! some ASCII art or markdown tables, you would need to preserve LFs along with
+//! precise formatting. You can ask `structopt` to skip preprocessing step
+//! via `#[structopt(no_mangle_doc)]` attribute. **This attribute
+//! must be applied to each field separately**, there's no global switch.
+//!
+//! Keep in mind that `structopt` will *still remove one leading space from each
+//! line*, even if this attribute is present.
```

Comment:
Keep in mind that `structopt` will *still remove one leading space from each line*, even if this attribute is present, to allow for a space between `///` and the content.

Reply:
Maybe "to account for  a space between /// and the content"; it's not like we allowing or disallowing anything.

Reply:
I've never tried it, but doesn't Rust allow `///ugly comment`?

As a native English speaker, I think that "account for" sounds a little too much like it's treating the space after `///` as inevitable in this context.

"Allow" sounds like "If we didn't strip 0 or 1 leading spaces, we'd be *de facto* forcing a style on the user.